src/mono/wasm/runtime/cs-to-js.ts

```diff
@@ -69,7 +69,7 @@ function _unbox_mono_obj_root_with_known_nonprimitive_type_impl(root: WasmRoot<a
         case MarshalType.ARRAY_UINT:
         case MarshalType.ARRAY_FLOAT:
         case MarshalType.ARRAY_DOUBLE:
-            throw new Error("Marshalling of primitive arrays are not supported.  Use the corresponding TypedArray instead.");
+            throw new Error("Marshaling of primitive arrays are not supported.  Use the corresponding TypedArray instead.");
```

Comment:
If we remove the corresponding typed array support we'll need to update this error :) But as mentioned in another comment I think maybe we keep a subset of it.